File: Assignment5/Sprite.java
import java.awt.Graphics;

abstract class Sprite
{
    int x, y, w, h;
    boolean isCoinBlock;

    abstract void update();
    abstract void draw(Graphics g);
    abstract Json Marshal();

    void coinCounter() {};

    boolean isCoin()
    {
        return false;
    }

    boolean isBrick()
    {
        return false;
    }

    boolean isMario()
    {
        return false;
    }
}

File: Assignment7/Sprite.js
class Sprite {
	constructor(x, y, w, h, model) {
		// basic structure of sprites including a model reference
		this.x = x;
		this.y = y;
		this.w = w;
		this.h = h;
		this.model = model;
		this.image = new Image();


		// setup types
		this.isMario = false;
		this.isCoin = false;
		this.isBrick = false;
		this.isCoinBlock = false;
	}

	update() {}

	draw() {}
}